quantinuumqircheck/api.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020-2024 Quantinuum
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+check for QIR module if it is compatible with quantinuum devices
+"""
+
+from typing import Union
+
+import pyqir
+
+from .qircheck import validate_qir_base
+
+
+def qircheck(qir: Union[str, bytes, pyqir.Module]) -> bool:
```

Comment:
See comment on sigature of `validate_qir_base`; same applies here.

Reply:
Done in [change return type](https://github.com/CQCL/quantinuum-qircheck/pull/1/commits/4ae4a1bde25db153767785b67a86a3441e12de16)
[4ae4a1b](https://github.com/CQCL/quantinuum-qircheck/pull/1/commits/4ae4a1bde25db153767785b67a86a3441e12de16)